source/extensions/filters/http/cache/http_cache.h

```diff
@@ -279,7 +279,7 @@ class HttpCacheFactory : public Config::TypedFactory {
   // etc.
   virtual std::shared_ptr<HttpCache>
   getCache(const envoy::extensions::filters::http::cache::v3::CacheConfig& config,
-           Server::Configuration::FactoryContext& context) PURE;
+           Server::Configuration::CommonFactoryContext& context) PURE;
```

Comment:
I think that means your implementation couldn't work with per-route config, since [Server::Configuration::ServerFactoryContext](https://github.com/envoyproxy/envoy/blob/main/envoy/server/factory_context.h#L152-L180) (which is what you get in the per-route config constructor) doesn't have `processContext`.